# Configuration for the StateEstimatorNode

# Primary state estimation method
primary: 'ema'  # Default: 'ukf7d'
# Choices: ['ema', 'ukf2d', 'ukf7d', 'ukf12d', 'mocap', 'simulator']

# Other state estimation methods to run alongside the primary one
others: 
  - 'ema'
  - 'mocap'
# Default: []
# Choices: ['ema', 'ukf2d', 'ukf7d', 'ukf12d', 'mocap', 'simulator']

# Boolean flags for using throttled topics
ir_throttled: False  # Default: False

imu_throttled: False  # Default: False

optical_flow_throttled: False  # Default: False

camera_pose_throttled: False  # Default: False

# Number of spatial dimensions in which to simulate the drone's motion
sdim: 1  # Default: 1
# Choices: [1, 2, 3]

# Use the Student's t-distribution-based UKF
student_ukf: False  # Default: False

# IR sensor variance for 1D simulation
ir_var: 0.0  # Default: None

# Frequency to run the predict-update loop of the UKF
loop_hz: 30.0  # Default: 30.0
